zk/stages/stage_interhashes.go

```diff
@@ -179,6 +171,7 @@ func SpawnZkIntermediateHashesStage(s *stagedsync.StageState, u stagedsync.Unwin
 
 		if root != expectedRootHash {
 			eridb.RollbackBatch()
+			tx.Rollback()
```

Comment:
This means that "tx" object could not be used anymore in this stage loop right? If yes, then are we sure that there isn't anything that could write something to the db like progress updates, etc?